### PR TITLE
BUG: Correct seasonal order

### LINKED ISSUE #7893: ETSModel.fit() reads initial_seasonal array in the wrong order.
#### Describe the bug

When I fit my ETSModel it looks as if fit() reads the initial_seasonal array in the wrong order. This makes the seasonal component of the fitted model look strange and the resulting forecast is not accurate.

In the example below there is two years of data (24 months) with very similar pattern between year one and two. Even so the forecast is not good and the model fails to capture the seasonal pattern. When you look at the details you see that the inital_seasonal component of the ETS model looks fine but when you look at the season component in the fitted model it seem to be wrong. More precisely, if you look at the first twelve months of the season component it looks like a mirror image of the inital_seasonal values. This makes me suspect that the .fit() method reads the array in the wrong order.

#### Code Sample, a copy-pastable example if possible
```python
from statsmodels.tsa.exponential_smoothing.ets import ETSModel as ETS
import matplotlib.pyplot as plt
import pandas as pd
import numpy as np

time_series = [
                23939.4,
                26092.0, 
                30088.1, 
                18531.1, 
                21384.0, 
                19387.5,
                0.0,
                0.0,
                0.0,
                0.0,
                0.0, 
                7229.3,
                27747.5,
                31152.0, 
                35640.0, 
                28433.4, 
                29304.0,
                23336.5,
                4378.0,
                0.0,
                0.0,
                0.0,
                0.0,
                1526.6
            ]

first = pd.Series(time_series[:12])
last = pd.Series(time_series[-12:])
time_series = pd.Series(time_series)

model = ETS(
    endog=time_series,
    initialization_method='heuristic',
    error='add',
    seasonal='add',
    seasonal_periods=12
)
fitted = model.fit(
    full_output=False,
    disp=False,
)
forecast = fitted.forecast(12)

fig, (ax) = plt.subplots(1, 1, figsize=(11, 6))
ax = first.plot(color='blue', label='time_series first 12 months', title="Seasonal profiles")
ax = last.plot(ax=ax, label="time_series last 12 months", color='dodgerblue')
ax = pd.Series(model.initial_seasonal).plot(ax=ax, label='initial_seasonal', color='green')
ax = pd.Series(fitted.season[12:]).plot(ax=ax, label='first 12 months fitted.season', color='red')
_ = ax.legend()
plt.show()
```

#### Small gamma suggest that seasonal component shouldn't change too dramatically.
print(fitted.gamma)
#### Forecast, as you can see it get's it completely wrong.
fig, (ax) = plt.subplots(1, 1, figsize=(11, 6))
ax = time_series.plot(color='blue', label='time_series', title="Time series and forecast")
ax = pd.Series(forecast).plot(ax=ax, label='Forecast', color='red')
_ = ax.legend()
plt.show()

<details>

**Note**: As you can see, there are many issues on our GitHub tracker, so it is very possible that your issue has been posted before. Please check first before submitting so that we do not have to handle and close duplicates.

**Note**: Please be sure you are using the latest released version of `statsmodels`, or a recent build of `main`. If your problem has been fixed in an unreleased version, you might be able to use `main` until a new release occurs. 

**Note**: If you are using a released version, have you verified that the bug exists in the main branch of this repository? It helps the limited resources if we know problems exist in the current main branch so that they do not need to check whether the code sample produces a bug in the next release.

</details>

If the issue has not been resolved, please file it in the issue tracker.

#### Expected Output

I would expect the seasonal component of the fitted model to be more similar to the initial_seasonal values from the ETSModel.

#### Output of ``import statsmodels.api as sm; sm.show_versions()``

<details>

[paste the output of ``import statsmodels.api as sm; sm.show_versions()`` here below this line]
INSTALLED VERSIONS
------------------
Python: 3.9.7.final.0

statsmodels
===========

Installed: 0.13.0 (C:\Miniconda3\envs\forecast_engine_test\lib\site-packages\statsmodels)

Required Dependencies
=====================

cython: Not installed
numpy: 1.21.4 (C:\Miniconda3\envs\forecast_engine_test\lib\site-packages\numpy)
scipy: 1.7.1 (C:\Miniconda3\envs\forecast_engine_test\lib\site-packages\scipy)
pandas: 1.3.4 (C:\Miniconda3\envs\forecast_engine_test\lib\site-packages\pandas)
    dateutil: 2.8.2 (C:\Miniconda3\envs\forecast_engine_test\lib\site-packages\dateutil)
patsy: 0.5.2 (C:\Miniconda3\envs\forecast_engine_test\lib\site-packages\patsy)

Optional Dependencies
=====================

matplotlib: 3.4.3 (C:\Miniconda3\envs\forecast_engine_test\lib\site-packages\matplotlib)
    backend: module://matplotlib_inline.backend_inline 
cvxopt: Not installed
joblib: 1.1.0 (C:\Miniconda3\envs\forecast_engine_test\lib\site-packages\joblib)

Developer Tools
================

IPython: 7.29.0 (C:\Miniconda3\envs\forecast_engine_test\lib\site-packages\IPython)
    jinja2: Not installed
sphinx: Not installed
    pygments: 2.10.0 (C:\Miniconda3\envs\forecast_engine_test\lib\site-packages\pygments)
pytest: 6.2.5 (C:\Miniconda3\envs\forecast_engine_test\lib\site-packages\pytest)
virtualenv: Not installed

</details>

### PR DESCRIPTION
**Pull request by bashtage**
Reverse seasonal during model intialization

- [X] closes #7893
- [x] tests added / passed. 
- [X] code/documentation is well formatted.  
- [X] properly formatted commit message. See 
      [NumPy's guide](https://docs.scipy.org/doc/numpy-1.15.1/dev/gitwash/development_workflow.html#writing-the-commit-message). 

<details>

**Notes**:

* It is essential that you add a test when making code changes. Tests are not 
  needed for doc changes.
* When adding a new function, test values should usually be verified in another package (e.g., R/SAS/Stata).
* When fixing a bug, you must add a test that would produce the bug in main and
  then show that it is fixed with the new code.
* New code additions must be well formatted. Changes should pass flake8. If on Linux or OSX, you can
  verify you changes are well formatted by running 
  ```
  git diff upstream/main -u -- "*.py" | flake8 --diff --isolated
  ```
  assuming `flake8` is installed. This command is also available on Windows 
  using the Windows System for Linux once `flake8` is installed in the 
  local Linux environment. While passing this test is not required, it is good practice and it help 
  improve code quality in `statsmodels`.
* Docstring additions must render correctly, including escapes and LaTeX.

</details>

### CODE REVIEW

**Comment by bashtage** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|maintenance/0.13.x|[](https://github.com/statsmodels/statsmodels/pull/8055)|


### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)
